### ISSUE TITLE
Releases with non-preID names but with pre_id not being renamed.

### ISSUE THREAD

**Issue by gkoh** [Murodese/pynab#268]
Not sure if my expectations are wrong, but I've got some releases in category 5030 and 5040 that are not being renamed despite having a pre_id.
The pre_id, when I check in the pres table has all the right information.

Let me know what I need to track this further, I'm not very familiar with the requests and postprocessing side of things.

**Comment by brookesy2** [comment]
Is this still happening? I haven't noticed it on my end.

**Comment by gkoh** [comment]
Sorry, I've been away for a few weeks.
Since then my prebot stopped scraping pres for reasons unknown, so I don't know if it still happens.
Let me fix that up and I'll update.

**Comment by brookesy2** [comment]
Ok, you may notice some NoneType errors, which are just warnings it couldn't parse something. Been busy with work, but I need to submit a pull request to make the error more friendly, as well as spend some time in the chan to see why it isn't parsed.

**Comment by gkoh** [comment]
OK, before I confirm, I need to understand the renaming process.

I have an entry in the release table where:
name = blah.123.hdtv
search_name = blah 123 hdtv
pre_id = 11111
group_id = 93 (a.b.etc)

In the pres table I have:
id = 11111
name = Blah.S01E23.HDTV
searchname = Blah S01E23 HDTV

Should I expect the release table names to be updated during postprocessing or are my expectations totally wrong?

Actually, looking at the code in requests.py, it looks to only rename from an entry with a name that is a request ID. Am I reading that right?

**Comment by gkoh** [comment]
OK, here's a concrete example from a recent scrape:

```
# select added,name,search_name,pre_id from releases where pre_id = 365326 order by added desc;
           added           |       name        |    search_name    | pre_id 
---------------------------+-------------------+-------------------+--------
 2016-04-20 05:58:40.19295 | regret-menace.mkv | regret menace mkv | 365326

# select id,name,searchname from pres where id = 365326;
   id   |                                                   name                                                   |                                                searchname                                                
--------+----------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------
 365326 | Dont.Be.a.Menace.To.South.Central.While.Drinking.Your.Juice.In.The.Hood.1996.iNTERNAL.DVDRip.x264-REGRET | Dont Be a Menace To South Central While Drinking Your Juice In The Hood 1996 iNTERNAL DVDRip x264 REGRET
```

From my reading of the request postprocessing and API search code, it doesn't look like this type of pre is being handled.

**Comment by brookesy2** [comment]
That is correct, it will only rename if it is a request ID.

**Comment by gkoh** [comment]
Thanks for confirming, I've updated the issue title to reflect this finding.
So, how can we go about supporting this?
For me it's primarily an issue with a number of TV show posts that title the season and episode as '123' instead of 'S01E23'.

**Comment by brookesy2** [comment]
Majority of cases like that should be handled by postproc parsing/renaming.

https://github.com/Murodese/pynab/blob/master/pynab/requests.py I believe?

I'm not as familiar with this area, but potentially could add in smarter renaming.

Edit: Fixed the link.

**Comment by gkoh** [comment]
I've been trawling over the various ways pynab fixes releases and I think this should be handled in the same way as the request ID pres.
The only real difference is the name of the release isn't numeric.
The interesting part is how to track and/or fix releases that have already been processed.
It looks like the numeric-named pres move from category 8010 which essentially marks them as 'processed', whereas the majority of these alphanumeric-pres are already in the right category.
Anything else I'm missing? Other suggestions?

**Comment by brookesy2** [comment]
You can always make a one off script and add it to the scripts dir, will pick up any old releases marked as processed. Then just build the renaming into the existing structure.

**Comment by gkoh** [comment]
I've just committed 6bd037e to hopefully address this. The actual solution was far simpler than I had originally thought, pynab/requests.py already does most of the right things.
On my install, running scripts/rename_pres.py fixed over 1500 releases (interestingly quite a few were named 'REQ: xxxxx' and therefore didn't get renamed before).

Please take a look and let me know. If it all looks OK, I'll submit a pull.

**Comment by brookesy2** [comment]
That is awesome news! Will give it a whirl and see. @Murodese will need to merge the PR though.

**Comment by gkoh** [comment]
Thanks!
I should note that after running scripts/rename_pres.py, the next postprocess loop will clean up the fixed releases.

**Comment by gkoh** [comment]
This was fixed via #278, closing.
